### PR TITLE
Adjust container fanout example

### LINKED ISSUE #722: Dynamic Fanout Container Example not properly passing parameters
<h2>Pre-bug-report checklist</h2>

**1. This bug can be reproduced using YAML**
- [ ] Yes [👉 Please report a bug to the Argo Workflows GitHub 👈](https://github.com/argoproj/argo-workflows/issues/new/choose)
- [x] No

**2. This bug occurs when...**
- [ ] running Hera code without submitting to Argo (e.g. when exporting to YAML)
- [x] running Hera code and submitting to Argo

<h2>Bug report</h2>

**Describe the bug**
When attempting to run [dynamic_fanout_container.py](https://github.com/argoproj-labs/hera/blob/c7374e76e4ed07432f00d7bafeed8dd4c563b9f8/examples/workflows/dynamic_fanout_container.py), the fanout task properly fans out to 3 separate tasks, but does not properly accept the value parameter. 

What the parameter is input as:
image

**How to fix the example**
I am unclear if Hera was implemented incorrectly, or the example was just not completely updated from [the v4 version](https://github.com/argoproj-labs/hera/blob/e93b0b0d972e658c00f8c786314fc47f26ae64f9/examples/dynamic_fanout_container.py). I was able to correct the example and get it working properly, but I am not sure if that is the _best_ solution.

To have the example run correctly, [the fanout task](https://github.com/argoproj-labs/hera/blob/c7374e76e4ed07432f00d7bafeed8dd4c563b9f8/examples/workflows/dynamic_fanout_container.py#L16C1-L21C2) should be changed to:
```diff
fanout = Container(
    name="fanout",
-   inputs=[Parameter(name="value", value="{{item.value}}")],
+   inputs=[Parameter(name="value")],
+   arguments=[Parameter(name="value", value="{{item.value}}")],
    image="alpine:latest",
    command=["echo", "{{inputs.parameters.value}}"],
)
```
This will change [the yaml file](https://github.com/argoproj-labs/hera/blob/c7374e76e4ed07432f00d7bafeed8dd4c563b9f8/examples/workflows/dynamic-fanout-container.yaml) like so :
```diff
spec:
  entrypoint: d
  templates:
  - dag:
      tasks:
      - name: generate
        template: generate
      - depends: generate
+       arguments:
+         parameters:
+         - name: value
+           value: '{{item.value}}'
        name: fanout
        template: fanout
        withParam: '{{tasks.generate.outputs.result}}'
    name: d
  - container:
      command:
      - echo
      - '[{"value": "a"}, {"value": "b"}, {"value": "c"}]'
      image: alpine:latest
    name: generate
  - container:
      command:
      - echo
      - '{{inputs.parameters.value}}'
      image: alpine:latest
    inputs:
      parameters:
      - name: value
-       value: '{{item.value}}'
    name: fanout
```
Upon running the corrected version of the example:
image

**Environment**
 - OS: Amazon Linux 2
 - Browser: Firefox
 - Version of Hera: 5.5.2
 - Version of Argo: 3.4.8

### PR DESCRIPTION
**Pull request by flaviuvadan**
**Pull Request Checklist**
- [x] Fixes #722
- [ ] Tests added
- [ ] Documentation/examples added
- [ ] [Good commit messages](https://cbea.ms/git-commit/) and/or PR title

### CODE REVIEW

File: examples/workflows/upstream/suspend-template.upstream.yaml [hunk 1]
```diff
@@ -32,7 +32,7 @@ spec:
 
   - name: delay
     suspend:
-      duration: "20"    # Must be a string. Default unit is seconds. Could also be a Duration, e.g.: "2m", "6h", "1d"
```
Comment (dejamiko) [inline examples/workflows/upstream/suspend-template.upstream.yaml:35]: Is this change intentional?

**Review by dejamiko — COMMENTED** [review]
Looks good, just one question

File: examples/workflows/upstream/suspend-template.upstream.yaml (hunk 1 above)
Comment (samj1912) [reply examples/workflows/upstream/suspend-template.upstream.yaml:35]: It's synced directly from upstream so we don't control it.

**Review by samj1912 — APPROVED** [review]
